dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/Amazon/AmazonIOService.cs

```diff
@@ -0,0 +1,213 @@
+                var responseBody = JsonSerializer.Deserialize<TitanTextResponse>(reader.ReadToEnd());
+                var textContents = new List<TextContent>();
+
+                if (responseBody?.Results != null && responseBody.Results.Count > 0)
+                {
+                    string? outputText = responseBody.Results[0].OutputText;
+                    textContents.Add(new TextContent(outputText));
+                }
+                return textContents;
+            }
+        }
+    }
+    /// <summary>
+    /// Builds the ConverseRequest object for the Bedrock ConverseAsync call with request parameters required by Amazon Titan.
+    /// </summary>
+    /// <param name="modelId">The model ID.</param>
+    /// <param name="chatHistory">The messages between assistant and user.</param>
+    /// <param name="settings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    public ConverseRequest GetConverseRequest(string modelId, ChatHistory chatHistory, PromptExecutionSettings settings)
+    {
+        var titanRequest = new TitanRequest.TitanChatCompletionRequest
+        {
+            Messages = chatHistory.Select(m => new Message
+            {
+                Role = MapRole(m.Role),
+                Content = new List<ContentBlock> { new() { Text = m.Content } }
+            }).ToList(),
+            System = new List<SystemContentBlock>(), // { new SystemContentBlock { Text = "You are an AI assistant." } },
+            InferenceConfig = new InferenceConfiguration
+            {
+                Temperature = this.GetExtensionDataValue(settings.ExtensionData, "temperature", 0.7f),
+                TopP = this.GetExtensionDataValue(settings.ExtensionData, "topP", 0.9f),
+                MaxTokens = this.GetExtensionDataValue(settings.ExtensionData, "maxTokenCount", 512),
+            },
+            AdditionalModelRequestFields = new Document(),
+            AdditionalModelResponseFieldPaths = new List<string>()
+        };
+        var converseRequest = new ConverseRequest
+        {
+            ModelId = modelId,
+            Messages = titanRequest.Messages,
+            System = titanRequest.System,
+            InferenceConfig = titanRequest.InferenceConfig,
+            AdditionalModelRequestFields = titanRequest.AdditionalModelRequestFields,
+            AdditionalModelResponseFieldPaths = titanRequest.AdditionalModelResponseFieldPaths,
+            GuardrailConfig = null, // Set if needed
+            ToolConfig = null // Set if needed
+        };
+        return converseRequest;
+    }
+
+    private TValue GetExtensionDataValue<TValue>(IDictionary<string, object>? extensionData, string key, TValue defaultValue)
+    {
+        if (extensionData == null || !extensionData.TryGetValue(key, out object? value))
+        {
+            return defaultValue;
+        }
+
+        if (value is TValue typedValue)
+        {
+            return typedValue;
+        }
+
+        return defaultValue;
```

Comment:
This could be simplified
```
private static TValue GetExtensionDataValue<TValue>(IDictionary<string, object>? extensionData, string key, TValue defaultValue)
{
    if (extensionData?.TryGetValue(key, out object? value) == true && value is TValue typedValue)
    {
        return typedValue;
    }

    return defaultValue;
}
```